docs/network-layer.md

```diff
@@ -0,0 +1,70 @@
+# Network Layer
+This document describes the network layer of the Concordium blockchain.
+
+## Overview 
+The network layer consists of a peer-to-peer (P2P) network. Each node in the network is connected to a *set of peers*. The node uses the established connections for various tasks including: *retrieving more peers*, _consensus_, _catching up_ on the chain etc.
```

Comment:
Maybe also add that there are two kinds of peers here, the bootstrappers and "node" (or however they are called in the code). 
And bootstrappers don't run consensus, they just serve peers. Bootstrappers are mentioned below, but they could be in this overview as well.